lib/ui/app_starter/app_starter.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_survey/database/shared_preferences.dart';
+import 'package:flutter_survey/di/di.dart';
+import 'package:flutter_survey/ui/home/home_screen.dart';
+import 'package:flutter_survey/ui/login/login_screen.dart';
+
+class AppStarter extends StatelessWidget {
+  AppStarter({super.key});
+
+  final _sharedPreferences = getIt<SharedPreferencesStorage>();
+
+  @override
+  Widget build(BuildContext context) {
+    return _sharedPreferences.isLoggedIn
```

Comment:
@Shayokh144 

Would it be possible to check if `accessToken` is stored instead (from secure storage)?

Then we don't need `shared_preferences_foundation` 💭

Reply:
@Tuubz to check if a key is exist or not in secure_storage we need to make a [async call](https://github.com/mogol/flutter_secure_storage/issues/338) but here we need the result immediately otherwise we need to use [FutureBuilder](https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html) in that case we need to show a loading screen when we are waiting for the result of secure_storage's async function.

Reply:
Interesting, thank you for the explanation 🙏